content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc. 
+
+On top of this, devs are more important to their companies than ever. Devs have gone from being viewed as cost centers to being seen as the center of product development, so the issue of complexity is a business problem just as much as a dev problem.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you learn things *about* code? Google went from helping you find things *on* the web (matching web pages), to helping you learn things *from* the web (information and answers), and we're doing the same with code.
+
+This is what we call a code intelligence platform, and compared to code search, it's more helpful, more times per day. **A code intelligence platform helps devs learn and understand information about their code so they can operate more effectively, with a deeper understanding of their codebase and how it works.** 
+
+A code intelligence platform should persist data beyond the code itself and make that data accessible. Here are some examples of what we see the platform doing in the future:
+
+- Trying to figure out the best way to call a function? Code search could show you a semi-exhaustive list of call sites. Code intelligence is also being able to see how the code owners of that function themselves have recently been calling that function.
+- Need to debug an issue in code? Code search could help you navigate within the same project to understand it. Code intelligence is also being able to see what changed and why, and trace calls across projects and dependencies/dependents.
+- Want to fix or improve something across your entire codebase? Code search might help you find where needs fixing. Code intelligence is also being able to push up a quick fix everywhere, track its progress, and monitor for regressions. Sourcegraph does this today.
+- Need to see how your team's API is being used? Code search could show you a list of file locations. Code intelligence is also seeing which other teams are using it and the trend of usage over time.
+
+A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Sourcegraph is committed to being universal so you can get full code intelligence *and* keep using the best tools in each category.
```

Comment:
Don't say etc. - instead, elaborate what we pull in information from. 

I also think we should be clear about what is on our roadmap and what is hypothetical.

Reply:
```suggestion
A code intelligence platform includes code search *and* information about your code from so many other tools that you rely on. Today, Sourcegaph pulls in information on code history, the dependency graph, and repository metadata. 
```

Reply:
Can you go back through this and delete every "also" and get right to the active part of the sentence? Can say it like "Code intelligence means you can now see how the code owners of that function themselves have recently been calling that function."

Reply:
> I also think we should be clear about what is on our roadmap and what is hypothetical.

Non-answer: we're still forming that FY24 roadmap, and should be wary of committing to any dates or defined features. That said, these are all ideas on our roadmap and at some point in time we may do all of them (may be infinite in the future point in time, of course), so they are all on the roadmap if you extended the roadmap 10 years. 

Up to others what we want to share given that constraint.